### PR TITLE
[Form] fix populating single widget time view data with different timezones

### LINKED ISSUE #46137: [Form] `TimeType` with a `reference_date`, `view_timezone` and `model_timezone` with a `widget` to `single_text` is not displaying correctly with `HTML5` enabled.
### Symfony version(s) affected

6.0.0 - 6.0.8

### Description

Hello folks!

When creating a `TimeType` form field with the `widget` as `single_text` and timezones the displayed value is not properly formatted for the browser.

```php
$fakeEntity = new \stdClass();
$fakeEntity->time = new \DateTime();

// Create the form field
->add('time', TimeType::class, [
    'html5' => true,
    'widget' => 'single_text',
    'view_timezone' => 'Europe/Paris',
    'model_timezone' => 'UTC',
    'reference_date' => new \DateTime('now', new \DateTimeZone('UTC')),
])
```

is outputting in HTML:
```html
<input type="time" id="form_time" name="form[time]" required="required" class="shadow-sm focus:ring-blue focus:border-blue-500 block w-full sm:text-sm border-gray-300 rounded-md" value="2022-04-21 14:49">
```

### How to reproduce

```php
$fakeEntity = new \stdClass();
$fakeEntity->time = new \DateTime();

// Create the form field
->add('time', TimeType::class, [
    'html5' => true,
    'widget' => 'single_text',
    'view_timezone' => 'Europe/Paris',
    'model_timezone' => 'UTC',
    'reference_date' => new \DateTime('now', new \DateTimeZone('UTC')),
])
```

is outputting in HTML:
```html
<input type="time" id="form_time" name="form[time]" required="required" class="" value="2022-04-21 14:49">
```

### Possible Solution

My guess is here: `form/Extension/Core/Type/TimeType.php:84`

We are dealing with the `reference_date` on the `PRE_SUBMIT` event but it's maybe too soon? The date shouldn't be displayed yet with the option `HTML5` enabled.

### Additional Context

image
> Here a result from my project with the reproduced step.

image
> And the result if I remove the date from the `value` field.

image
> And the result if I set `HTML5` to false.

Many thanks & best regards 😊

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #46137
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/TimeType.php [hunk 1]
```diff
@@ -85,7 +87,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 });
             }
 
-            $builder->addViewTransformer(new DateTimeToStringTransformer($options['model_timezone'], $options['view_timezone'], $format));
+            $viewTransformer = new DateTimeToStringTransformer($options['model_timezone'], $options['view_timezone'], $format);
+
+            if (null !== $parseFormat) {
+                $property = new \ReflectionProperty($viewTransformer, 'parseFormat');
+                $property->setAccessible(true);
+                $property->setValue($viewTransformer, $parseFormat);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/TimeType.php]: Does that mean we're missing a constructor argument to DateTimeToStringTransformer maybe?

File: src/Symfony/Component/Form/Extension/Core/Type/TimeType.php [hunk 2]
```diff
@@ -85,7 +87,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 });
             }
 
-            $builder->addViewTransformer(new DateTimeToStringTransformer($options['model_timezone'], $options['view_timezone'], $format));
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Extension/Core/Type/TimeType.php:88]:
```php
$builder->addViewTransformer(new DateTimeToStringTransformer(
    $options['model_timezone'],
    $options['view_timezone'],
    $parseFormat ?? $format
));
```
?

Comment (HeahDude) [reply src/Symfony/Component/Form/Extension/Core/Type/TimeType.php:88]: Ah no ok I see now

**Review by HeahDude — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]
(I updated the PR to add the parse format to the constructor)

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Comment by RSickenberg** [comment]
Thank you for the PR!

File: src/Symfony/Component/Form/Extension/Core/Type/TimeType.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/TimeType.php]: I didn't add it on purpose as that felt like introducing a new extension point and thus a new feature to me. But I am fine with the change you did as well.

**Pull request merged**